apps/prairielearn/src/pages/instructorGradebook/instructorGradebook.ts

```diff
@@ -84,22 +86,27 @@ router.get(
       throw new HttpStatusError(403, 'Access denied (must be a student data viewer)');
     }
 
-    if (req.params.filename === csvFilename(res.locals)) {
-      const assessmentsResult = await sqldb.queryAsync(sql.course_assessments, {
-        course_instance_id: res.locals.course_instance.id,
-      });
-      const userScoresCursor = await sqldb.queryCursor(sql.user_scores, {
+    if (req.params.filename === buildCsvFilename(res.locals)) {
+      const assessments = await queryRows(
+        sql.course_assessments,
+        { course_instance_id: res.locals.course_instance.id },
+        CourseAssessmentRowSchema,
+      );
+      const userScoresCursor = await queryCursor(sql.user_scores, {
         course_id: res.locals.course.id,
         course_instance_id: res.locals.course_instance.id,
       });
 
       const stringifier = stringifyStream({
         header: true,
-        columns: ['UID', 'UIN', 'Name', 'Role', ...assessmentsResult.rows.map((a) => a.label)],
-        transform(record) {
-          const score_percs = _.map(record.scores, (s) => s.score_perc);
-          return [record.uid, record.uin, record.user_name, record.role].concat(score_percs);
-        },
+        columns: ['UID', 'UIN', 'Name', 'Role', ...assessments.map((a) => a.label)],
+        transform: (record: GradebookRow) => [
+          record.uid,
+          record.uin,
+          record.user_name,
+          record.role,
+          ...assessments.map((a) => record.scores[a.assessment_id].score_perc ?? null),
```

Comment:
As implied above, this no longer relies on the `scores` array having the same order as the `assessments` query result.